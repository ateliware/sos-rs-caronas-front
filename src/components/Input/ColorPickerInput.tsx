import { useRef } from 'react';

import { Icon } from '../Icon';

import Input, { Props as InputProps } from './Input';

export type Props = {
  onSelect: (color: String) => void;
  colorSelected: string;
};

export default function ColorPickerInput(props: InputProps & Props) {
  const ref = useRef<HTMLInputElement>(null);

  return (
    <>
      <Input
        mask="color"
        prefixes={
          <div
            className="color-selected"
            style={{ backgroundColor: props.colorSelected }}
          />
        }
        suffixes={
          <div
            className="pr-s-100 m-s-50 mt-s-100 pointer"
            onClick={() => {
              ref.current?.focus();
              ref.current?.click();
            }}
          >
            <input
              type="color"
              ref={ref}
              onChange={(event) => {
                props.onSelect(event.target.value);
              }}
              tabIndex={-1}
              style={{ visibility: 'hidden', display: 'hidden' }}
              className="hidden"
            />
            <Icon>colorize</Icon>
          </div>
        }
        {...props}
      />
    </>
  );
}
