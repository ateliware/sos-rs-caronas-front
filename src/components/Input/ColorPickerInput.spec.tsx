import { fireEvent, render } from '@testing-library/react';

import ColorPickerInput from './ColorPickerInput';

describe('ColorPickerInput', () => {
  it('should render correctly', () => {
    const { container } = render(
      <ColorPickerInput onSelect={() => {}} colorSelected="#000" />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Should show color in the input`s prefix correctly', () => {
    const { container } = render(
      <ColorPickerInput onSelect={() => {}} colorSelected="#000" />
    );

    const color = container.querySelector(
      '.form-input__addons__container > div'
    );

    expect(color).toHaveStyle('background-color: #000');
  });

  it('Should show colorpicker modal correctly', () => {
    const { container } = render(
      <ColorPickerInput onSelect={() => {}} colorSelected="#000" />
    );

    const colorButton = container.querySelector(
      '.form-input__addons__addon--right > div'
    );

    const colorInput = container.querySelector(
      '.form-input__addons__addon--right > div > input'
    );

    fireEvent.click(colorButton!);

    expect(colorInput).toHaveFocus();
  });
});
