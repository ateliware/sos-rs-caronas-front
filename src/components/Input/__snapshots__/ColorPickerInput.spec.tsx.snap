// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPickerInput should render correctly 1`] = `
<div
  class="form-input__container"
>
  <div
    class="form-input__addons__container"
  >
    <div
      class="color-selected"
      style="background-color: rgb(0, 0, 0);"
    />
    <input
      class="form-input"
      id="input-:r0:"
      value=""
    />
    <div
      class="form-input__addons__addon form-input__addons__addon--right"
    >
      <div
        class="pr-s-100 m-s-50 mt-s-100 pointer"
      >
        <input
          class="hidden"
          style="visibility: hidden; display: hidden;"
          tabindex="-1"
          type="color"
        />
        <span
          class="material-symbols-rounded"
        >
          colorize
        </span>
      </div>
    </div>
  </div>
</div>
`;
